v2/pkg/controller/mpi_job_controller.go

```diff
@@ -1412,61 +1413,58 @@ func workerReplicas(job *kubeflow.MPIJob) int32 {
 	return 0
 }
 
-func podSSHAuthVolume(jobName string) ([]corev1.Volume, []corev1.VolumeMount) {
-	return []corev1.Volume{
-			{
-				Name: sshAuthVolume,
-				VolumeSource: corev1.VolumeSource{
-					Secret: &corev1.SecretVolumeSource{
-						SecretName:  jobName + sshAuthSecretSuffix,
-						DefaultMode: newInt32(0660),
-						Items: []corev1.KeyToPath{
-							{
-								Key:  corev1.SSHAuthPrivateKey,
-								Path: sshPrivateKeyFile,
-							},
-							{
-								Key:  sshPublicKey,
-								Path: sshPublicKeyFile,
-							},
-							{
-								Key:  sshPublicKey,
-								Path: sshAuthorizedKeysFile,
-							},
-						},
-					},
+func setupSSHOnPod(podSpec *corev1.PodSpec, job *kubeflow.MPIJob) {
+	podSpec.Volumes = append(podSpec.Volumes,
+		corev1.Volume{
+			Name: sshAuthVolume,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: job.Name + sshAuthSecretSuffix,
+					Items:      sshVolumeItems,
 				},
 			},
-			{
-				Name: sshHomeVolume,
-				VolumeSource: corev1.VolumeSource{
-					EmptyDir: &corev1.EmptyDirVolumeSource{},
-				},
+		},
+		corev1.Volume{
+			Name: sshHomeVolume,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
 			},
-		}, []corev1.VolumeMount{
+		})
+
+	mainContainer := &podSpec.Containers[0]
+	mainContainer.VolumeMounts = append(mainContainer.VolumeMounts,
+		corev1.VolumeMount{
+			Name:      sshHomeVolume,
+			MountPath: job.Spec.SSHAuthMountPath,
+		})
+
+	// The init script sets the permissions of the ssh folder in the user's home
+	// directory. The ownership is set based on the security context of the
+	// launcher's first container.
+	launcherSecurityCtx := job.Spec.MPIReplicaSpecs[kubeflow.MPIReplicaTypeLauncher].Template.Spec.Containers[0].SecurityContext
+	initScript := "" +
+		"cp -RL /mnt/ssh/* /mnt/home-ssh && " +
+		"chmod 700 /mnt/home-ssh && " +
+		"chmod 600 /mnt/home-ssh/*"
+	if launcherSecurityCtx != nil && launcherSecurityCtx.RunAsUser != nil {
+		initScript += fmt.Sprintf(" && chown %d -R /mnt/home-ssh", *launcherSecurityCtx.RunAsUser)
+	}
+	podSpec.InitContainers = append(podSpec.InitContainers, corev1.Container{
+		Name:  "init-ssh",
+		Image: "alpine:3.14",
```

Comment:
could we make this image configurable? some cluster environment is isolated from docker registry.

Reply:
Done